### ISSUE TITLE
OMEMO implementation state

### ISSUE THREAD

**Issue by COM8** [COM8/UWP-XMPP-Client#5]
This issue tracks the state of the OMEMO implementation.

--> 5%

**Comment by COM8** [comment]
Yes, good point!
Have to remember myself to keep it simple :D

**Comment by COM8** [comment]
That's the current state. Is this simple enough, what do you think?

![img](https://user-images.githubusercontent.com/11741404/49034522-830dea00-f1b2-11e8-8978-017e90c24d87.png)

**Comment by Echolon** [comment]
Hey - cool - I like it! Yes, now the user directly understands what it about. You should add the selection 'Unencrypted', too. Maybe a toggle thing is nice too.

And of course the lock symbol, but I guess it's work in progress.

:raised_hands:

**Comment by Supded** [comment]
Fix fingerprint representation, it must be conversation capable.

**Comment by Supded** [comment]
![31 01 2019_13 05 43 7972 2](https://user-images.githubusercontent.com/22551225/52050966-e02f6e80-255a-11e9-81e5-e2e0e09f0a71.jpg)
![31 01 2019_13 07 59 0300](https://user-images.githubusercontent.com/22551225/52050967-e02f6e80-255a-11e9-9f1d-d921f1267711.png)

**Comment by COM8** [comment]
Yes. I know what's going wrong and a fix will be published with the UI overhaul.

**Comment by COM8** [comment]
> 
> 
> ![31 01 2019_13 05 43 7972 2](https://user-images.githubusercontent.com/22551225/52050966-e02f6e80-255a-11e9-81e5-e2e0e09f0a71.jpg)
> ![31 01 2019_13 07 59 0300](https://user-images.githubusercontent.com/22551225/52050967-e02f6e80-255a-11e9-9f1d-d921f1267711.png)

40d7a8b fixes it. Finally!

**Comment by stevenroose** [comment]
Is the progress list at the top still up to date? No encryption in MUCs yet?

**Comment by COM8** [comment]
Yes, still up to date since for MUC OMEMO support a few other bugs and milestones like:
* #50 
* #55

have to be fixed/archived first.

**Comment by Neustradamus** [comment]
XEP-0384: OMEMO Encryption has been updated to 0.4.0:
- http://www.aptest.com/standards/htmldiff/htmldiff.pl?oldfile=https://xmpp.org/extensions/attic/xep-0384-0.3.0.html&newfile=https://xmpp.org/extensions/attic/xep-0384-0.4.0.html

**Comment by COM8** [comment]
Already created an Issue for it #126 .
Thanks for the hint!

**Comment by Neustradamus** [comment]
XEP-0384: OMEMO Encryption has been updated to 0.5.0:
- https://xmpp.org/extensions/xep-0384.html
- http://www.aptest.com/standards/htmldiff/htmldiff.pl?oldfile=https://xmpp.org/extensions/attic/xep-0384-0.4.0.html&newfile=https://xmpp.org/extensions/attic/xep-0384-0.5.0.html
- http://www.aptest.com/standards/htmldiff/htmldiff.pl?oldfile=https://xmpp.org/extensions/attic/xep-0384-0.3.0.html&newfile=https://xmpp.org/extensions/attic/xep-0384-0.4.0.html

PS : Not sure that 0.3.0 was complete before.

**Comment by closewall** [comment]
Any updates on this?

**Comment by COM8** [comment]
Sorry for the late response, but YES!
UWPX now supports ([OMEMO 0.7.0 (2020-09-05)](https://xmpp.org/extensions/xep-0384.html)) and I rewrote the complete OMEMO lib from the ground up.
https://github.com/UWPX/UWPX-Client/releases/tag/v.0.31.0.0
